pandas/core/frame.py

```diff
@@ -10435,22 +10439,60 @@ def apply(
         0  1  2
         1  1  2
         2  1  2
+
+        Advanced users can speed up their code by using a Just-in-time (JIT) compiler
+        with ``apply``. The main JIT compilers available for pandas are Numba and Bodo.
+        In general, JIT compilation is only possible when the function passed to
+        ``apply`` has type stability (variables in the function do not change their
+        type during the execution).
+
+        >>> import bodo
+        >>> df.apply(lambda x: x.A + x.B, axis=1, engine=bodo.jit(parallel=True))
+
+        Note that JIT compilation is only recommended for functions that take a
+        significant amount of time to run. Fast functions are unlikely to run faster
+        with JIT compilation.
         """
-        from pandas.core.apply import frame_apply
+        if engine is None or isinstance(engine, str):
+            from pandas.core.apply import frame_apply
 
-        op = frame_apply(
-            self,
-            func=func,
-            axis=axis,
-            raw=raw,
-            result_type=result_type,
-            by_row=by_row,
-            engine=engine,
-            engine_kwargs=engine_kwargs,
-            args=args,
-            kwargs=kwargs,
-        )
-        return op.apply().__finalize__(self, method="apply")
+            if engine is None:
+                engine = "python"
```

Comment:
Don't you need to test for strings that are not `"python"` or `"numba"` ?

Reply:
Yes, good catch, thank you. I thought it was checked in `apply.py` as the other parameters, but this was missing. Added, and also added a test for it.